src/snapshot.c

```diff
@@ -574,6 +574,7 @@ static void clientSessionRDBLoad(RedisModuleIO *rdb)
         unsigned long long id = RedisModule_LoadUnsigned(rdb);
         client_session->client_id = id;
         client_session->local = false;
+        client_session->client = RedisModule_CreateModuleClient(rr->ctx);
```

Comment:
What about rdbSave? don't we want to save the client state too? (dirty state for example)

Reply:
done now

Reply:
If missing, maybe we can add a test for snapshot & snapshot delivery. Just to hit those lines where we save/restore sessions